xbmc/addons/AddonManager.cpp

```diff
@@ -879,11 +879,13 @@ bool CAddonMgr::LoadAddonDescription(const std::string &directory, AddonPtr &add
 
 bool CAddonMgr::AddUpdateRuleToList(const std::string& id, AddonUpdateRule updateRule)
 {
+  m_events.Publish(AddonEvents::UpdateRuleModified(id));
```

Comment:
Question: Is it correct to fire the event before calling `m_updateRules.AddUpdateRuleToList` or should this actualyl be done after that call? Logically, I would say that is has to be fired after calling `m_updateRules.AddUpdateRuleToList`, not before.

Reply:
As publishing events is async, this might work, but by luck, I'd say.

Reply:
as i see it now, i think the correct call would be (to respect the db-error handling)
```
if (m_updateRules.AddUpdateRuleToList(m_database, id, updateRule))
{
  m_events.Publish(AddonEvents::UpdateRuleModified(id));
  return true;
}
return false;
```
but don't know if this is safer with `async` publishing, how can we do this correctly?

Reply:
Yes. That’s more correct.

Reply:
Yes, this seems correct. Same fix logic applies for the other method that was modified by this PR.

Reply:
BTW, for async publishing it is not important when the message will arrive, but it should be send only if data actually have changed.

Reply:
ok, sorry i was a bit too fast with merging this ;)

Reply:
All good. Will you come up with a fix?

Reply:
yes, sure. will come shortly